controller/token.go

```diff
@@ -126,6 +126,26 @@ func (c *TokenController) Generate(ctx *app.GenerateTokenContext) error {
 			ProviderType:          account.KeycloakIDP,
 			RegistrationCompleted: true,
 		}
+
+		err = transaction.Transactional(c.app, func(tr transaction.TransactionalResources) error {
+			// Using the old-fashioned service
+			err := tr.Identities().Save(ctx, &devIdentity)
+			if err != nil {
+				return err
+			}
+			err = tr.Users().Save(ctx, &devIdentity.User)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err": err,
+			}, "failed to create a user and identity for dev user")
+			return jsonapi.JSONErrorResponse(ctx, err)
```

Comment:
you should return an `InternalServerError` with a message like `"failed to create a user and identity for dev user"`, otherwise, it's the repository error that will be returned to the user, which is not what we may want.